### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by mkusher** [mkusher/padawan.php#42]
Padawan could not generate index for `Symfony\Component\VarDumper\Tests\Fixtures\NotLoadableClass`

**Comment by mkusher** [comment]
The problem is that padawan tries to recursively add links to parent classes. And when parent is the same class it goes to an infinite recursion.

**Comment by mkusher** [comment]
a good test case by @jludwig

``` php
<?php

class C extends C {}


class E extends F {}
class F extends E {}

class G extends H {}
class H extends I {}
class I extends G {}
```

**Comment by allan-simon** [comment]
maybe it was already obvious, but the workaround for symfony projects, is to simply open the file containing `Symfony\Component\VarDumper\Tests\Fixtures\NotLoadableClass`  and commenting it , generate the index and uncommenting it

**Comment by dakira** [comment]
So, this is blocking re-integration into PHPCompanion. Any chance this can be fixed? Maybe the reflection code of Symfony itself (that introduced this testfile) helps? https://github.com/symfony/var-dumper/commit/f133ef94df600e5fa7f7d26761c7f663c25ed040

**Comment by mkusher** [comment]
@dakira https://github.com/mkusher/padawan.php/pull/54 I started working on it sometime ago, but haven't got a chance to finish it

**Comment by okdewit** [comment]
Personally, I'd love an `--ignore` option of some kind, like ctags provides. Some way to include/exclude directories and files, or maybe a class blacklist.

**Comment by mkusher** [comment]
@okdewit yeah, it has been discussed https://github.com/mkusher/padawan.php/issues/20
